pkg/kudoctl/packages/resolver/resolver_incluster.go

```diff
@@ -0,0 +1,43 @@
+package resolver
+
+import (
+	"fmt"
+
+	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/convert"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+)
+
+// InClusterResolver resolves packages that are already installed in the cluster on the client-side. Note, that unlike
+// other resolvers, the resulting 'packages.Package' struct does not contain package 'packages.Files' (we don't have
+// the original files).
+type InClusterResolver struct {
+	c  *kudo.Client
+	ns string
+}
+
+func (r InClusterResolver) Resolve(name string, appVersion string, operatorVersion string) (*packages.Package, error) {
+	ovn := kudoapi.OperatorVersionName(name, operatorVersion)
+
+	ov, err := r.c.GetOperatorVersion(ovn, r.ns)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve operator version %s/%s:%s", r.ns, ovn, appVersion)
+	}
+
+	o, err := r.c.GetOperator(name, r.ns)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve operator %s/%s", r.ns, name)
+	}
+
+	i := convert.BuildInstanceResource(name, operatorVersion)
```

Comment:
Why do we create a "dummy" instance here? The other InClusterResolver just makes a note that Instance is empty (which makes sense), i'm not sure why we create an non-existing instance here?

Reply:
> Why do we create a "dummy" instance here?

that's the same way the _soon-to-be-installed_ `Instance` resource is created when package files are resolved. Except that we don't need to create `O/OV` - they are taken from the cluster.

> The other InClusterResolver just makes a note that Instance is empty

the other/server-side resolver is used to make sure that all dependencies exist in the cluster - the `Instance` resource is not needed there as it will be created later by the `KudoOperatorTask`.

Reply:
Ah, thanks, that makes sense.